src/main/java/com/synopsys/integration/alert/web/api/role/RoleActions.java

```diff
@@ -86,9 +94,25 @@ public UserRoleModel createRole(RolePermissionModel rolePermissionModel) throws
         return userRoleModel;
     }
 
-    public UserRoleModel updateRole(Long roleId, RolePermissionModel rolePermissionModel) throws AlertDatabaseConstraintException, AlertConfigurationException {
+    public UserRoleModel updateRole(Long roleId, RolePermissionModel rolePermissionModel) throws AlertDatabaseConstraintException, AlertConfigurationException, AlertFieldException {
         String roleName = rolePermissionModel.getRoleName();
-        authorizationUtility.updateRoleName(roleId, roleName);
+        Optional<AlertFieldStatus> roleNameMissingError = validateRoleNameFieldRequired(roleName);
+        if (roleNameMissingError.isPresent()) {
+            throw AlertFieldException.singleFieldError(roleNameMissingError.get());
+        }
+
+        UserRoleModel existingRole = getExistingRoleOrThrow404(roleId);
+        boolean targetRoleNameIsUsedByDifferentRole = authorizationUtility.getRoles()
+                                                          .stream()
+                                                          .filter(role -> !role.getId().equals(existingRole.getId()))
+                                                          .anyMatch(role -> role.getName().equalsIgnoreCase(roleName));
```

Comment:
More of a thought, but would this go through the stream twice? Would it be more efficient to put both checks in either the filter call or the anyMatch call?

Reply:
It makes no difference: 
https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html